File: src/main/java/net/benwoodworth/fastcraft/core/api/gui/GuiApi.java
package net.benwoodworth.fastcraft.core.api.gui;

import net.benwoodworth.fastcraft.core.api.dependencies.Dependencies;
import org.jetbrains.annotations.NotNull;

/**
 * Created by bwoodworth on 12/13/2016.
 */
public class GuiApi<TItem> {

    @NotNull
    private final Dependencies<TItem> dependencies;

    public GuiApi(@NotNull Dependencies<TItem> dependencies) {
        this.dependencies = dependencies;
    }
}

File: src/main/kotlin/net/benwoodworth/fastcraft/core/api/gui/GuiApi.kt
package net.benwoodworth.fastcraft.core.api.gui

import net.benwoodworth.fastcraft.core.api.dependencies.Dependencies

/**
 * An API for creating [Gui]'s
 */
class GuiApi<TItem>(
        /** The plugin dependencies */
        val dependencies: Dependencies<TItem>
)
